keycloak/sms-provider/src/main/java/org/sunbird/keycloak/rest/RequiredActionLinkProvider.java

```diff
@@ -0,0 +1,212 @@
+
+    try {
+      ExecuteActionsActionToken token = new ExecuteActionsActionToken(user.getId(), expiration,
+          requiredActionList, redirectUri, clientId);
+      UriBuilder builder = LoginActionsService.actionTokenProcessor(session.getContext().getUri());
+      builder.queryParam(Constants.KEY,
+          token.serialize(session, session.getContext().getRealm(), session.getContext().getUri()));
+
+      String link = builder.build(session.getContext().getRealm().getName()).toString();
+      
+      Map<String, Object> response = new HashMap<>();
+      response.put(Constants.LINK, link);
+      return Response.ok(response).build();
+    } catch (Exception e) {
+      return ErrorResponse.error(Constants.ERROR_CREATE_LINK, Status.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private UserModel getUserByUserName(String userName) {
+    logger.debug("RestResourceProvider: getUserByUserName called");
+    if (StringUtils.isBlank(userName)) {
+      throw new WebApplicationException(ErrorResponse.error(
+          MessageFormat.format(Constants.ERROR_MANDATORY_PARAM_MISSING, userName, Constants.USERNAME),
+          Status.BAD_REQUEST));
+    }
+    UserModel user = KeycloakModelUtils.findUserByNameOrEmail(session,
+        session.getContext().getRealm(), userName);
+    
+    if (user == null) {
+      throw new WebApplicationException(ErrorResponse.error(
+          MessageFormat.format(Constants.ERROR_INVALID_PARAMETER_VALUE, userName, Constants.USERNAME),
+          Status.BAD_REQUEST));
+    }
+
+    if (!user.isEnabled()) {
+      throw new WebApplicationException(
+          ErrorResponse.error(Constants.ERROR_USER_IS_DISABLED, Status.BAD_REQUEST));
+    }
+    
+    return user;
+  }
+
+  private List<String> getRequiredActionList(String actionName) {
+    if (StringUtils.isBlank(actionName)) {
+      throw new WebApplicationException(
+          ErrorResponse.error(MessageFormat.format(Constants.ERROR_MANDATORY_PARAM_MISSING, actionName,
+              Constants.REQUIRED_ACTION), Status.BAD_REQUEST));
+    }
+    
+    List<String> requiredActionList = new ArrayList<>();
+    switch (actionName) {
+      case Constants.UPDATE_PASSWORD:
+        requiredActionList.add(UserModel.RequiredAction.UPDATE_PASSWORD.name());
+        break;
+      case Constants.VERIFY_EMAIL:
+        requiredActionList.add(UserModel.RequiredAction.VERIFY_EMAIL.name());
+        break;
+      default:
+        throw new WebApplicationException(
+            ErrorResponse.error(MessageFormat.format(Constants.ERROR_INVALID_PARAMETER_VALUE, actionName,
+                Constants.REQUIRED_ACTION), Status.BAD_REQUEST));
+    }
+    
+    return requiredActionList;
+  }
+  
+  private int getExpirationInSecs(String expirationInSecsStr) {
+    int expirationInSecs;
+    
+    try {
+      if (StringUtils.isNotBlank(expirationInSecsStr)) {
+        expirationInSecs = Integer.parseInt(expirationInSecsStr);        
```

Comment:
Is there a maximum expiration time to enforce?

Reply:
yes max expiration time we can enforce is INT_MAX as the maximum 'int' value